src/textual/_animator.py

```diff
@@ -93,9 +95,19 @@ class SimpleAnimation(Animation):
     final_value: object
     easing: EasingFunction
     on_complete: CallbackType | None = None
+    animate_on_level: AnimationsEnum = AnimationsEnum.BASIC
+    """Minimum level required for the animation to take place (inclusive)."""
+
+    def __post_init__(self) -> None:
+        """Cache baseline that determines whether an animation runs."""
+        # We cache this value so we don't have to `.get` the context var repeatedly.
+        self.show_animations = active_app.get().show_animations
 
     def __call__(self, time: float) -> bool:
-        if self.duration == 0:
+        if (
```

Comment:
This is where we need the `app.show_animations`. Could we pass it to `__call__` ?

Alternatively, do we even need to do this logic here? The animator calls this method. Could we not perform this logic from the animator? The animator has a reference to the app.

Reply:
We _could_ perform the logic there but I think it's cleaner here.
If we do it on the animator, then we also need to set the attribute to its value in the animator and I think it'll look clunky there.

Reply:
You may need to elaborate on "cleaner" and "clunky" which are quite subjective terms.

It does feel like it is the animators job to decide if to animate something. Perhaps an Animation object should grow a `finalize` method which sets the attribute to its final stage.